src/gg.regression.unity.bots/Runtime/Scripts/StateRecorder/BotSegments/Models/BotSegment.cs

```diff
@@ -232,5 +234,170 @@ public void WriteToStringBuilder(StringBuilder stringBuilder)
+                    var segment = ParseSegment(result.Item2, fileName);
+                    if (segment != null)
+                    {
+                        results.Add(segment);
+                    }   
+                }
+
+                // recurse into the next level of directories
+                results = results.Concat(
+                    LoadSegmentsInDirectory(directory)
+                ).ToList();
+            }
+
+            return results;
+        }
+        
+        /**
+         * <summary>
+         * Recursively look through directories for Segement files, and load them (in a build of the game)
+         * </summary>
+         * <param name="path">
+         * Directory to search for Segments
+         * </param>
+         */
+        private static List<BotSequenceEntry> LoadSegmentsInDirectoryForRuntime(string path)
+        {
+            var results = new List<BotSequenceEntry>();
+            var directories = Directory.EnumerateDirectories(path);
+            foreach (var directory in directories)
+            {
+                var jsons = Resources.LoadAll(directory, typeof(TextAsset));
+                foreach (var jsonObject in jsons)
+                {
+                    try
+                    {
+                        var json = (jsonObject as TextAsset)?.text ?? "";
+                        var segment = ParseSegment(json, jsonObject.name);
+
+                        // don't add segments with duplicate names
+                        if (results.Any(s => s.entryName == segment.entryName))
+                        {
+                            continue;
+                        }
+                        
+                        results.Add(segment);
+                    }
+                    catch (Exception e)
+                    {
+                        throw new Exception($"Exception reading a Segment json file from resource path: {directory}", e);
+                    }
+                }
+
+                results = results.Concat(
+                    LoadSegmentsInDirectoryForRuntime(directory)
+                ).ToList();
+            }
+
+            return results;
+        }
+        
+        /**
+         * <summary>
+         * Parses a json string as either a Segment or SegmentList
+         * </summary>
+         * <param name="json">
+         * JSON string to parse
+         * </param>
+         * <param name="fileName">
+         * File the Segment or SegmentList is derived from
+         * </param>
+         */
+        private static BotSequenceEntry ParseSegment(string json, string fileName)
+        {
+            try
+            {
+                var entry = new BotSequenceEntry
+                {
+                    path = fileName,
+                };
+
+                try
+                {
+                    var segment = JsonConvert.DeserializeObject<BotSegmentList>(json);
+                    entry.description = segment.description;
```

Comment:
We should check the version support when loading these files in.  Segments/SegmentLists/etc

```csharp
                if (segment.EffectiveApiVersion > SdkApiVersion.CURRENT_VERSION)
                {
                    throw new Exception($"Bot segment list file contains a segment which requires SDK version {segment.EffectiveApiVersion}, but the currently installed SDK version is {SdkApiVersion.CURRENT_VERSION}");
                }
```

Reply:
I've added a check for both the Segment and SegmentList types. I've compared like `segment.EffectiveApiVersion >= SdkApiVersion.CURRENT_VERSION`, I think equal versions is ok

Reply:
This should be >

We are checking the bad case where the sequence/segment/etc is TOO new for the current runtime.